homeassistant/components/bluetooth/base_scanner.py

```diff
@@ -83,26 +78,43 @@ def __init__(
         self.scanning = True
         self._last_detection = 0.0
         self._start_time = 0.0
-        self._cancel_watchdog: CALLBACK_TYPE | None = None
+        self._cancel_watchdog: asyncio.TimerHandle | None = None
+        self._loop: asyncio.AbstractEventLoop | None = None
+
+    @hass_callback
+    def async_setup(self) -> CALLBACK_TYPE:
+        """Set up the scanner."""
+        self._loop = asyncio.get_running_loop()
+        return self._unsetup
 
     @hass_callback
     def _async_stop_scanner_watchdog(self) -> None:
         """Stop the scanner watchdog."""
         if self._cancel_watchdog:
-            self._cancel_watchdog()
+            self._cancel_watchdog.cancel()
             self._cancel_watchdog = None
 
     @hass_callback
     def _async_setup_scanner_watchdog(self) -> None:
         """If something has restarted or updated, we need to restart the scanner."""
         self._start_time = self._last_detection = MONOTONIC_TIME()
         if not self._cancel_watchdog:
-            self._cancel_watchdog = async_track_time_interval(
```

Comment:
I removed usage of async_track_time_interval here since this code will eventually live in a new library but I'm doing this in steps to make sure I don't break anything